### PR TITLE
Promisify

### PR DESCRIPTION
**Pull request by wearhere**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -100,21 +102,20 @@ Object.assign(Lock.prototype, {
         return {1, index, ttl};
       `;
 
-    this._scripty.loadScript('acquireScript', acquireScript, (err, script) => {
-      if (err) return done(err);
+    const scriptPromise = deferred();
+    this._scripty.loadScript('acquireScript', acquireScript, scriptPromise.defer());
```
Comment (wearhere) [inline src/index.js]: tfw `node-redis-scripty` has not promisified yet: 😿

**Comment by wearhere** [comment]
https://github.com/mixmaxhq/redfour/pull/21/files?w=1

File: src/index.js [hunk 2]
```diff
@@ -181,58 +182,61 @@ Object.assign(Lock.prototype, {
...
+          // Try one last time and return whatever the acquireLock returns
+          if (!acquiring) {
+            return tryAcquire();
+          }
+        }, waitTtl);
+      }
+
+      // try to acquire a lock
+      tryAcquire();
+    });
```
Comment (skeggse) [inline src/index.js]: it feels like there should be a better async/await-native representation for this but that's also effort and this should do the trick

Comment (wearhere) [reply src/index.js]: yeah i was wondering about that… I guess the repeated calls to `tryAcquire` could be unrolled into a loop maybe

File: src/index.js [hunk 3]
```diff
@@ -151,21 +153,20 @@ Object.assign(Lock.prototype, {
         return {0, "conflict", data.index};
       `;
 
-    this._scripty.loadScript('releaseScript', releaseScript, (err, script) => {
-      if (err) return done(err);
+    const scriptPromise = deferred();
+    this._scripty.loadScript('releaseScript', releaseScript, scriptPromise.defer());
+    const script = await scriptPromise;
```
[the remark this answers is not in the document]
Comment (wearhere) [reply src/index.js]: Yeah thanks for the suggestion. I would have promisified if I could do that in one place, just thought it unfortunate to have to promisify each returned script.

File: src/index.js [hunk 4]
```diff
@@ -181,58 +182,61 @@ Object.assign(Lock.prototype, {
    *    different processes trying to acquire this lock.
    * @param {Number} ttl Automatically release acquired lock after TTL (ms). Must be positive integer
    * @param {Number} waitTtl Give up until ttl (in ms) or wait indefinitely if value is 0
-   * @param {Function} done Callback
+   *
+   * @return {Promise<Error, Lock>}
```
[the remark this answers is not in the document]
Comment (wearhere) [reply src/index.js]: oh right

**Pull request merged**